src/dpo/src/Optdp.cpp

```diff
@@ -563,41 +592,45 @@ void Optdp::createNetwork() {
       if (termMap_.end() != it_p) {
         n = it_p->second->getId();  // The node id.
 
-        if (nw_->getNode(n)->getId() != n || nw_->getEdge(e)->getId() != e) {
+        if (network_->getNode(n)->getId() != n || network_->getEdge(e)->getId() != e) {
+          logger_->error(DPO, 104, "Improper terminal indexing while connecting pins.");
           ++errors;
         }
 
-        Pin* ptr = nw_->createAndAddPin(nw_->getNode(n),nw_->getEdge(e));
+        Pin* ptr = network_->createAndAddPin(network_->getNode(n),network_->getEdge(e));
 
         // These don't need an offset.
         ptr->setOffsetX(0.0);
         ptr->setOffsetY(0.0);
         ptr->setPinHeight(0.0);
         ptr->setPinWidth(0.0);
-
-        // XXX: Not correct, but okay for now!
-        ptr->setPinLayer(0);
+        ptr->setPinLayer(0); // Set to zero since not currently used.
 
         ++p;  // next pin.
       } else {
+        logger_->error(DPO, 105, "Could not find node for terminal while connecting pins.");
         ++errors;
       }
     }
 
     ++e;  // next edge.
   }
   if (e != nEdges) {
+    logger_->error(DPO, 106, "Unexpected total edge count.  Expected {:d}, but got {:d}",
+        nEdges, e);
     ++errors;
   }
   if (p != nPins) {
+    logger_->error(DPO, 107, "Unexpected total pin count.  Expected {:d}, but got {:d}",
+        nPins, p);
     ++errors;
   }
 
   if (errors != 0) {
```

Comment:
Note that the logger's error method both prints the message but also throws an exception.  So there is no need to count errors as this will throw on the 1st error.